text/0000-decouple-dependencies.md

```diff
@@ -0,0 +1,403 @@
+update metadata, and package metadata into compressed archives and images. For
+buildpack users, we’ll create some tooling to fetch metadata, overlay metadata
+from multiple sources, pre-cache assets from metadata, and feed metadata and
+pre-cached assets into `pack build`.
+
+### Library Enhancement
+The next step will be to add new functionality in `libpak` and `packit` for
+consuming metadata from the defined metadata location and for consuming
+dependency assets from this metadata. This can be done in an additive way so
+that existing functionality continues to be supported. Support for
+`buildpack.toml` dependency metadata can be deprecated and removed on a
+separate timeline.
+
+### Update Buildpacks
+When we have library support, we’ll be able to update the Paketo buildpacks
+that are installing dependencies to use the metadata provided. For example, we
+can update the Bellsoft Liberica or Amazon Corretteo buildpacks to install
+their JRE/JDK dependencies using this new metadata.
+
+We propose a flag of `BP_EXTERNAL_METADATA_ENABLED` which defaults to `false`
+for use as buildpacks are being converted. In the default state, this flag
+tells a buildpack to use the metadata included with buildpack.toml. When set to
+`true`, a buildpack should use the new metadata. This can provide a way for
+users to test the new functionality without impacting existing users.
+
+### Write Documentation
+Documentation will be added at each step. Initial tool documentation will be on
+the repositories for the tools. The `libpak` and `packit` libraries will update
+their docs as support is added. As individual buildpacks are updated, the
+`README.md` for each buildpack will be updated with the newly supported
+options. When an entire language family is complete, the information will be
+merged into the official Paketo documentation.
+
+### Buildpack Migration Process
+Once a language family has added support for the new metadata format and
+documentation has been updated, the language family may begin the deprecation
+process. This proposal does not dictate how that process should be done, but
+provides some recommendations. The language family team ultimately has
+discretion on how to proceed.
+
+The recommendation of this proposal is to announce the change in the release
+notes and on Slack, providing links to documentation of the new feature. The
+goal of this migration is that there is no loss of functionality for buildpack
+users, so we do not need to trigger a major release and we can remove the older
+functionality and buildpack.toml entries as is convenient for buildpack teams.
+
+There is some risk in removing buildpack.toml entries. We have not documented
+the format of the dependency information but the format has been consistent for
+the current lifetime of the buildpacks and it is possible users may have tools
+or integrations to read it. Teams may want to continue maintaining both sets of
```

Comment:
Commercial vendors definitely read the current metadata from `buildpack.toml` for outcomes like provenance/source tracking.

As long as the format is the same, I'd expect this tooling to continue to work with minimal changes. But it won't be zero effort for this tooling to continue working.

Reply:
Yes, I suspect all sorts of use cases will pop up. Good communication will be key to working through this in a way that's fair to everyone.